### ISSUE TITLE
Increase minimum required python version to 3.10

### ISSUE THREAD

**Issue by Artur-Galstyan** [patrick-kidger/equinox#774]
Given that since JAX 0.4.31, the minimum required Python version is 3.10, shouldn't this repo also increase the minimum Python version too?

> jax 0.4.31
> Changes
> 
> The minimum Python version is now 3.10. 3.10 will remain the minimum supported version until July 2025.

(Which also means we could use the union operator for types `|` :D)

**Comment by lockwo** [comment]
Wouldn't that also mean bumping the jax dependency (a fair amount, from jax>=0.4.13)?

**Comment by patrick-kidger** [comment]
Yeah, I don't see a reason to bump this until we also find ourselves needing to depend on JAX 0.4.31+.
